### PR TITLE
Fix `LicensePage` too much spacing padding when `applicationVersion` and  `applicationLegalese` are empty

### LINKED ISSUE #99559: Open source licenses page has too much spacing by default
This is a nitpick, but the Material licenses screen has an unusual amount of spacing at the top. Perhaps this is only true when apps do not set all the optional description text properties, but some major apps do this, so it'd be great to handle it well.

Attached is a screenshot of the GPay licenses screen which uses a very default setup.

 
Internal issue number: 222333499

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/99559

<details> 
<summary>minimal code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Material App',
      home: Sample(),
    );
  }
}

class Sample extends StatelessWidget {
  const Sample({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return const Scaffold(
      body: Center(
        child: LicensePage(
          // applicationName: 'LicensePage test app',
          // applicationVersion: '0.1.2',
          applicationIcon: FlutterLogo(size: 100),
          // applicationLegalese: 'I am the very model of a modern major general.',
        ),
      ),
    );
  }
}
``` 
	

</details>

| Before | After |
| --------------- | --------------- |
|  |  |

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/about.dart [hunk 1]
```diff
@@ -493,17 +493,21 @@ class _AboutProgram extends StatelessWidget {
...
+          if (version != null && version != '')
+            Padding(
+              padding: const EdgeInsets.only(bottom: _textVerticalSeparation),
+              child: Text(
+                version,
+                style: Theme.of(context).textTheme.bodyText2,
+                textAlign: TextAlign.center,
+              ),
+            ),
+          if (legalese != null && legalese != '')
```
Comment (guidezpl) [inline packages/flutter/lib/src/material/about.dart:505]:
```suggestion
          if (legalese != '')
```

**Review by guidezpl — APPROVED** [review]
with small nits

File: packages/flutter/lib/src/material/about.dart [hunk 2]
```diff
@@ -493,17 +493,21 @@ class _AboutProgram extends StatelessWidget {
...
-            style: Theme.of(context).textTheme.bodyText2,
-            textAlign: TextAlign.center,
-          ),
-          const SizedBox(height: _textVerticalSeparation),
-          Text(
-            legalese ?? '',
-            style: Theme.of(context).textTheme.caption,
-            textAlign: TextAlign.center,
-          ),
+          if (version != null && version != '')
```
[the remark this answers is not in the document]
Comment (TahaTesser) [reply packages/flutter/lib/src/material/about.dart:496]: Thanks! This works great

File: packages/flutter/lib/src/material/about.dart (hunk 1 above)
Comment (TahaTesser) [reply packages/flutter/lib/src/material/about.dart:505]: Unlike `version `, `legalese` is nullable and making this change, fails the test with `Null check operator used on a null value`

Comment (guidezpl) [reply packages/flutter/lib/src/material/about.dart:505]: I think it should work the null check a few lines later is removed

**Pull request merged**